File: dev-docs/snippets/databases/connect.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Connect to a database and get a DoDb instance for ad-hoc record queries.
// JS-only — the Swift client has no record CRUD/query surface.
export function connect(client: JsBaoClient, databaseId: string) {
  // #region example
  const db = client.databases.connect(databaseId);
  // db now exposes record-level query/find/save/patch/count/aggregate.
  // #endregion example
  return db;
}

File: dev-docs/snippets/databases/connect.swift
import JsBaoClient

// Connect to a database and use the `DoDb` handle for ad-hoc record ops
// (save / find / query / count / aggregate + index management) — parity with
// the JS `databases.connect()` direct-record surface.
func useDoDb(client: JsBaoClient) async throws {
  // #region example
  let db = client.databases.connect(databaseId: "tasks")
  let id = try await db.save(modelName: "Task", data: [
    "title": "Review PR",
    "priority": 2,
  ])
  let row = try await db.find(modelName: "Task", id: id)
  // #endregion example
  _ = row
}
